src/Field.tsx

```diff
@@ -113,9 +127,28 @@ export function Field<T extends FieldProps>({
     ...props
   };
 
+  const renderInput = () => (
+    <React.Fragment>
+      {renderIconBefore && (
+        <div
+          className={join([theme.icon, theme.iconBefore])}
+          role="presentation"
+        >
+          {renderIconBefore()}
+        </div>
+      )}
+      {renderIcon && (
```

Comment:
Should this be after `render(inputProps)`?

Reply:
The position in the markup is irrelevant, since its presentational anyway. And positioning it here makes it significantly easier to position it with CSS.

I think there are probably are cases where we might need both, albeit rare. An icon on the left that signifies something about the field type, and one on the right to signify error state. Its a stretch I know, but thought it would make it more future proof.

Reply:
Okay, makes sense to me!

Reply:
See my comment about bootstrap though. If we do want to support their `input-group-prepend` and `input-group-append` we'd have to make this change. I'm not jazzed about it.